From c17f21ec5c9038a73a2792043ba8a09a906f3b29 Mon Sep 17 00:00:00 2001
From: PyScotch <pyscotch@noreply.invalid>
Date: Tue, 13 Oct 2026 16:21:43 +0000
Subject: [PATCH 229/240] Bugfix: suffix Scotch routine names in Fortran test
 template [report PyScotch]

When SCOTCH_NAME_SUFFIX is set, the Fortran wrappers of the library
are suffixed too, but "test_scotch_mesh_build_f.f90.in" called
`scotchfmeshinit()` and `scotchfmeshbuild()` by their plain names, so
that the test program could not be linked, with both the Makefile and
CMake build chains. Use the "##SCOTCH_NAME_SUFFIX" placeholder that
"dummysizes" substitutes when it processes the template.
---
 src/check/test_scotch_mesh_build_f.f90.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/check/test_scotch_mesh_build_f.f90.in b/src/check/test_scotch_mesh_build_f.f90.in
index 757a521..3868b35 100644
--- a/src/check/test_scotch_mesh_build_f.f90.in
+++ b/src/check/test_scotch_mesh_build_f.f90.in
@@ -39,7 +39,7 @@
 !*                the mesh building routine.              **
 !*                                                        **
 !*   DATES      : # Version 7.0  : from : 24 jan 2026     **
-!*                                 to     24 jan 2026     **
+!*                                 to     13 oct 2026     **
 !*                                                        **
 !234567*****************************************************
 
@@ -68,7 +68,7 @@ program test_scotch_mesh_build
 
   ierr = 0
 
-  call scotchfmeshinit(meshdat, ierr)
+  call scotchfmeshinit##SCOTCH_NAME_SUFFIX(meshdat, ierr)
 
   velmbas=1
   vnodbas=4
@@ -80,7 +80,7 @@ program test_scotch_mesh_build
   verttab = [1,5,9,13,14,16,18,20,21,22,23,25]
   edgetab = [5,11,7,6,10,5,11,4,8,9,6,7,2,2,1,1,3,1,3,3,3,2,2,1]
 
-  call scotchfmeshbuild(meshdat, & ! doubleprecision (*) meshdat
+  call scotchfmeshbuild##SCOTCH_NAME_SUFFIX(meshdat, & ! doubleprecision (*) meshdat
   velmbas, & ! integer*num velmbas,
   vnodbas, & ! integer*num vnodbas,
   velmnbr, & ! integer*num velmnbr,
-- 
2.47.3

